darshan-util/darshan-mpiio-logutils.c

```diff
@@ -518,6 +554,22 @@ static void darshan_log_agg_mpiio_files(void *rec, void *agg_rec, int init_flag)
                 }
                 break;
             case MPIIO_F_FASTEST_RANK_TIME:
+
+                if(!shared_file_flag || mpi_rec->base_rec.rank == -1)
```

Comment:
At first glance, I'm not sure the second part of the if statement is needed here (`mpi_rec->base_rec.rank == -1`). If the `rank == -1` then there are 2 case:
- there are multiple files represent in the input set of logs, which would be handled by the first if condition (`!shared_file_flag`)
- there is a single file represented in the input set of logs, where ideally we could just preserve the FASTEST/SLOWEST counters rather than setting to `-1` here